include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -471,6 +471,81 @@ def NVMMASharedEncodingAttr :
   let hasCustomAssemblyFormat = 1;
 }
 
+def SwizzledBlocksSharedEncodingAttr :
+  TritonGPU_Attr<"SwizzledBlocksSharedEncoding", "swizzled_blocks_shared_encoding", [SharedEncodingTrait, LayoutEncodingTrait]> {
+  let mnemonic = "swizzled_blocks_shared";
+
+  let description = [{
+This shared encoding is similar to SwizzledSharedEncodingAttr,
+but instead of repeating swizzling pattern every `blockSize = maxPhase*perPhase` rows of memory object,
+this layout changes swizzling pattern.
+
+Let's consider an element with position = (row, col).
+For simplicity, we do not vectorize memory,
+i.e. vec == 1 and layout swizzles inidividual elements.
+
+Swizzled coordinates within memory object are (row, swizzledCol):
+
+  phase   = (row / perPhase) % maxPhase
+  blockNo = (row / perPhase) / maxPhase % maxPhase
```

Comment:
Writing it as `row / (perPhase * maxPhase) % maxPhase` might be more easy to interpret.